### ISSUE TITLE
Dock remains visible even when some apps are full screen (Hyprland specific?)

### ISSUE THREAD

**Issue by anoraktrend** [nwg-piotr/nwg-dock-hyprland#12]
**Describe the bug**
Much like the "always invisible bug" on sway if the dock is launched before other applications, the inverse is true with hyprland. When you have a fullscreen application, the dock doesn't disappear and it doesn't dodge windows. 

**To Reproduce**
have the dock in the hyprland.conf 

**Expected behavior**
the dock should hide to dodge fullscreen, low enough or tiled windows.

**Screenshots**
![2023-08-03-172128_hyprshot](https://github.com/nwg-piotr/nwg-dock/assets/19198652/5f9b537d-173b-46f1-b8d1-1cd72e51f454)

**Desktop (please complete the following information):**
Arch linux, defaults, using  the -bin from the aur

**Comment by anoraktrend** [comment]
yes, that would make sense. I do just use the dock, (simply to mimic wf-shell's look). Does nwg-dock work well there?

**Comment by nwg-piotr** [comment]
I haven't seen Wayfire for a long time. No, none of my docks will work there. nwg-dock uses sway IPC, nwg-dock-hyprland communicates with Hyprland sockets. Wayfire, when I last saw it, had no IPC. Also they have their own, cool shell, so no point in making another one.

By the way: in current dock versions use of the `-d` argument makes the dock always appear on the overlay layer. I've already deleted this part of code, testing now. Will probably release tonight.

**Comment by nwg-piotr** [comment]
Done.

**Comment by anoraktrend** [comment]
the other issue is that it doesn't hide at all, but i think that's more just me not having read the documentation

**Comment by anoraktrend** [comment]
It  was  -d, -hd is for autohide with an altered hotspot delay

**Comment by anoraktrend** [comment]
Might be related to xwayland, in that case.

**Comment by cusniwtt** [comment]
I have same issue with you guys. Dock not hide when screen is going to fullscreen. So I create helper script to check dock state (is unhide?hide?) and toggle it. Maybe this can help your issue. Just save into `dock.sh` and add in autostart of hyprland `exec-once ~/path/to/your/sh/dock.sh`

```bash
#!/bin/sh

# Init dock
nwg-dock-hyprland -i 36 -w 5 -nolauncher -p right -r -mr 10 &
# Init Status dock
DOCK=true

check_active() {
    CURRENT_STATE=$(hyprctl activewindow -j | jq ".fullscreen")    # 0=Not, 2=Full
    # Dock is ON
    if [ "$DOCK" = true ]; then
        if [ "$CURRENT_STATE" = "2" ]; then
            nwg-dock-hyprland  # ON -> OFF
            DOCK=false
        fi
    # Dock is OFF
    else
        if [ "$CURRENT_STATE" = "0" ] || [ "$CURRENT_STATE" = "null" ]; then
            nwg-dock-hyprland  # OFF -> ON
            DOCK=true
        fi
    fi
}

handle() {
    case $1 in
        activewindowv2*) check_active;;
        fullscreen*) check_active;;
        workspacev2*) check_active;;
    esac
}

socat -U - UNIX-CONNECT:$XDG_RUNTIME_DIR/hypr/$HYPRLAND_INSTANCE_SIGNATURE/.socket2.sock | while read -r line; do 
    handle "$line"; 
done

# Exit dock
pkill -f nwg-dock-hyprland
```

You need to install `jq` and `socat` to make this script working.
